### ISSUE TITLE
HDMI audio not recognized in 22.04 LTS, kernel 6.0.12

### ISSUE THREAD

**Issue by smiledra** [pop-os/pop#2751]
**Distribution (run `cat /etc/os-release`):**
```
 NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
```
pipewire:
  Installed: 0.3.63~1672359253~22.04~af47487
  Candidate: 0.3.63~1672359253~22.04~af47487
  Version table:
 *** 0.3.63~1672359253~22.04~af47487 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
        100 /var/lib/dpkg/status
     0.3.48-1ubuntu2 500
        500 http://apt.pop-os.org/ubuntu jammy-updates/main amd64 Packages
     0.3.48-1ubuntu1 500
        500 http://apt.pop-os.org/ubuntu jammy/main amd64 Packages
```

```
nvidia-driver-515:
  Installed: 515.65.01-1pop0~1666367711~22.04~b8c0232
  Candidate: 515.65.01-1pop0~1666367711~22.04~b8c0232
  Version table:
     515.86.01-0ubuntu0.22.04.1 500
        500 http://apt.pop-os.org/ubuntu jammy-security/restricted amd64 Packages
        500 http://apt.pop-os.org/ubuntu jammy-updates/restricted amd64 Packages
 *** 515.65.01-1pop0~1666367711~22.04~b8c0232 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Issue/Bug Description:**
HDMI sound is not recognized in a fresh install. by selecting the correct channels, `aplay` is able to use HDMI speakers without problem, but `pipewire`, `pavucontrol`, and system setting > audio does not recognize the HDMI speaker. In particular, in pavucontrol > Configuration, the HDA controller is shown, but in Profile, the only selectable one is "off". 

I am using a setting with a laptop (w/ internal speaker), with an HDMI cable connecting to an external monitor. The internal speaker works fine, and the external speaker used to work in PopOS 21.04. The laptop is an OMEN Laptop 15-en0xxx with an AMD Ryzen 7 CPU and NVIDIA GeForce GTX 1660 Ti, running on NVIDIA graphics. The current linux kernel is 6.0.12-76060006-generic. 

**Steps to reproduce (if you know):**
Fresh install through USB > restart machine > HDMI is not recognized in Settings > audio. 

**Expected behavior:**
HDMI output is recognized and can be selected through Settings > audio. 

**Other Notes:**
- Internal speaker and bluetooth headsets works fine without problems. 
- Re-installing (or `systemctl restart`) of pipewire, etc. did not help. 
- Using other versions of nvidia (525, 470) did not make any difference. Purging and reinstalling nvidia also didn't help. 

Any troubleshooting pointers would also be appreciated!

**Comment by smiledra** [comment]
Update: since the pipewire update yeterday, I was able to select a profile for the TU 116 audio controller in `pavucontrol`. With some fidgeting I was able to get sound from the HDMI connection. The volume control is broken though -- although the slider goes up and down as I change the volume, there is no real effect on the sound. Also, in Pop's setting > Audio, I was not able to test individual speakers, unlike the non-HDMI sound outputs.

**Comment by Pablo1785** [comment]
Hey, I've seen your answer in a previous post about your issue. I believe I have the exact same problem.

Full context: 

HP Omen 15 Ryzen 7 4800h NVidia Geforce GTX 1660, Pop OS 22.04, just installed 2 days ago. I had 0 issues with HDMI sound in Pop OS 20.04. I tried multiple solutions, (reinstalling alsa, pipewire, changing things in pavucontrol), reinstalling the OS being the last one I tried. After reinstalling the problem was still there, so I searched some more and finally found out sound was muted in `alsa` thanks to this answer https://askubuntu.com/a/1415109

After unmuting and running the sound test command from the above answer I could finally hear sound coming out of my TV speakers.

So far this is the only time I could play any sounds through them. 

Selecting `Pro Mode` in `pavucontrol` for TU116 controller gives me six TU116 High Definition Audio Controller options in Settings > Sound (one without a number, the others numbered 7 through 11). I can see the sound meter under those moving correctly but no sound is coming out of my speakers.

Have you managed to solve this? Is Pop OS just unusable in its current state with this hardware?

**Comment by smiledra** [comment]
> Hey, I've seen your answer in a previous post about your issue. I believe I have the exact same problem.
> 
> Full context:
> 
> HP Omen 15 Ryzen 7 4800h NVidia Geforce GTX 1660, Pop OS 22.04, just installed 2 days ago. I had 0 issues with HDMI sound in Pop OS 20.04. I tried multiple solutions, (reinstalling alsa, pipewire, changing things in pavucontrol), reinstalling the OS being the last one I tried. After reinstalling the problem was still there, so I searched some more and finally found out sound was muted in `alsa` thanks to this answer https://askubuntu.com/a/1415109
> 
> After unmuting and running the sound test command from the above answer I could finally hear sound coming out of my TV speakers.
> 
> So far this is the only time I could play any sounds through them.
> 
> Selecting `Pro Mode` in `pavucontrol` for TU116 controller gives me six TU116 High Definition Audio Controller options in Settings > Sound (one without a number, the others numbered 7 through 11). I can see the sound meter under those moving correctly but no sound is coming out of my speakers.
> 
> Have you managed to solve this? Is Pop OS just unusable in its current state with this hardware?

Hi Pablo, 

I have reached a similar state. The `Pro Mode` for the TU116 controller appeared recently and I was able to select it. After selecting Pro Mode, there are some fidgeting I had to do with pavucontrol First, you need to know which number (7-11) your HDMI sound is coming through. For example, for mine it is 7. Then, you can play something on the application of your choice. When things are playing, in pavucontrol > Playback, you can see that application with a place where you can select the correct TU116 controller (i.e., `TU115 ... Controller Pro 7` for me). After that I was able to play something through this particular application, but things are still quite glitched for me: (1) the volume control keys can slide up and down but will not affect the actual HDMI volume, and (2) this has to be set manually for each application, so if I open a  new application, sound will default to some other channel and I have to manually set it to the correct TU116 output in pavucontrol. Perhaps one can write a script to automatize these and make it less broken, but I have decided to wait for future updates (I feel like things are getting closer though!).

**Comment by smiledra** [comment]
Additionally, `pavucontrol` closes sometimes a few seconds after I open it. In that case, I find it useful to keep opening it several times and eventually it wouldn't crash. I have no idea what happened.

**Comment by Pablo1785** [comment]
> Additionally, `pavucontrol` closes sometimes a few seconds after I open it. In that case, I find it useful to keep opening it several times and eventually it wouldn't crash. I have no idea what happened.

This happens to me too, though it seems to usually be the case after I actually change something in the settings, so nothing like that on e.x. first opening after a reboot.

Well I think I'll hold out for some updates. I would switch to Ubuntu but I really dislike the whole song and dance around reinstalling tools and software. Hopefully someone  working closer with the audio drivers will notice this, it really seems like a pipewire issue though

**Comment by smiledra** [comment]
> > Additionally, `pavucontrol` closes sometimes a few seconds after I open it. In that case, I find it useful to keep opening it several times and eventually it wouldn't crash. I have no idea what happened.
> 
> This happens to me too, though it seems to usually be the case after I actually change something in the settings, so nothing like that on e.x. first opening after a reboot.
> 
> Well I think I'll hold out for some updates. I would switch to Ubuntu but I really dislike the whole song and dance around reinstalling tools and software. Hopefully someone working closer with the audio drivers will notice this, it really seems like a pipewire issue though

Good to hear I'm not alone. I actually tried Ubuntu a month ago, and the HDMI output (visual, not even audio) is completely messed up. If we have very similar hardware, that might be a problem for you too. At least popos has slightly better NVIDIA (?) support. I also have an arch dual boot installed and the HDMI audio is completely broken there (not even recognized in alsa/aplay). That's actually why I installed popOS. Hopefully this will be fixed soon.

**Comment by feveal** [comment]
I also did a clean installation of Kubuntu 22.04 and the HDMI card was not detected. I did this twice with the same result. I tried all possible methods and that I was finding on the web. None worked.
In the end I decided to reinstall Kubuntu 20.04 from zero, in this versión I configured the sound outputs (20.04 detected all outputs) Then I upgraded to version 22.04 and now everything works.
![20 04 version](https://user-images.githubusercontent.com/62255440/223125545-f2eca192-9f7a-4b11-88df-79d8cd863b5f.png)
![22 04 versión after 20 04](https://user-images.githubusercontent.com/62255440/223125854-a3624f3a-d3bb-448c-9e47-2c6420ddca71.png)

**Comment by feveal** [comment]
The first image is the configuration in version 20.04 and the second from version 22.04 after the update from 20.04

**Comment by bassmanitram** [comment]
Promode appeared here - but even there I can't get sound through HDMI

**Comment by Pablo1785** [comment]
@smiledra I finally managed to get my HDMI working in linux again.

Following @feveal I decided to install Kubuntu 20.04. As expected, HDMI did not show up there either so I upgraded to 22.04. I started to have video issues in HDMI with multiple versions of NVidia drivers, mostly just no signal. 

I downgraded my NVidia drivers to 470 and the video was working just fine, but I was once again not able to select HDMI output anywhere, even though it showed up in alsa -l.

^All of the above was done on Linux kernel 5.17.

After upgrading to 6.3 (latest stable) I finally caught a glimpse of HDMI in my Audio settings. It was marked as inactive (profile: off) and generally didn't look right yet.

So I fiddled with multiple NVidia driver versions, trying out XOrg open source driver, 470, 515, 510.

What finally worked was 530 proprietary (not -open) driver version.

TL;DR - installed Kubuntu 22.04, upgraded kernel to 6.3 (latest stable), installed proprietary NVidia driver 530 (not -open, that one did not work).

I can only assume PopOS might also work after updating the kernel to latest stable and setting the driver version to latest proprietary, but I'm too fed up with this thing to try it.
